Repository: plone.outputfilters


Branch: refs/heads/master
Date: 2023-06-13T16:56:19-07:00
Author: David Glick (davisagli) <david@glicksoftware.com>
Commit: https://github.com/plone/plone.outputfilters/commit/c08c89ba0d67a6c5f52f7a52fe62551a83ee2318

fix error when resolveuid is called with no uuid

Files changed:
A news/43.bugfix
M plone/outputfilters/browser/resolveuid.py
M plone/outputfilters/tests/test_resolveuid_and_caption.py

b'diff --git a/news/43.bugfix b/news/43.bugfix\nnew file mode 100644\nindex 0000000..91b3d50\n--- /dev/null\n+++ b/news/43.bugfix\n@@ -0,0 +1 @@\n+Return a 404 Not Found response if the resolveuid view is called with no uuid. @davisagli\ndiff --git a/plone/outputfilters/browser/resolveuid.py b/plone/outputfilters/browser/resolveuid.py\nindex 8809af8..a781588 100644\n--- a/plone/outputfilters/browser/resolveuid.py\n+++ b/plone/outputfilters/browser/resolveuid.py\n@@ -43,6 +43,7 @@ def uuidFor(obj):\n class ResolveUIDView(BrowserView):\n     """Resolve a URL like /resolveuid/<uuid> to a normalized URL."""\n \n+    uuid = None\n     subpath = None\n \n     def publishTraverse(self, request, name):\ndiff --git a/plone/outputfilters/tests/test_resolveuid_and_caption.py b/plone/outputfilters/tests/test_resolveuid_and_caption.py\nindex 91e8120..403d3ee 100644\n--- a/plone/outputfilters/tests/test_resolveuid_and_caption.py\n+++ b/plone/outputfilters/tests/test_resolveuid_and_caption.py\n@@ -299,6 +299,10 @@ def test_resolveuid_view_bad_uuid(self):\n         res = self.publish("/plone/resolveuid/BOGUS")\n         self.assertEqual(404, res.status)\n \n+    def test_resolveuid_view_missing_uuid(self):\n+        res = self.publish("/plone/resolveuid")\n+        self.assertEqual(404, res.status)\n+\n     def test_resolveuid_view_subpath(self):\n         res = self.publish("/plone/resolveuid/%s/image_thumb" % self.UID)\n         self.assertEqual(301, res.status)\n'

Repository: plone.outputfilters


Branch: refs/heads/master
Date: 2023-06-14T10:57:28-07:00
Author: David Glick (davisagli) <david@glicksoftware.com>
Commit: https://github.com/plone/plone.outputfilters/commit/e9e38f41efc7a11632aaa449d6503fe9f248f096

Merge pull request #62 from plone/fix-resolveuid-missing-uuid

fix error when resolveuid is called with no uuid

Files changed:
A news/43.bugfix
M plone/outputfilters/browser/resolveuid.py
M plone/outputfilters/tests/test_resolveuid_and_caption.py

b'diff --git a/news/43.bugfix b/news/43.bugfix\nnew file mode 100644\nindex 0000000..91b3d50\n--- /dev/null\n+++ b/news/43.bugfix\n@@ -0,0 +1 @@\n+Return a 404 Not Found response if the resolveuid view is called with no uuid. @davisagli\ndiff --git a/plone/outputfilters/browser/resolveuid.py b/plone/outputfilters/browser/resolveuid.py\nindex 8809af8..a781588 100644\n--- a/plone/outputfilters/browser/resolveuid.py\n+++ b/plone/outputfilters/browser/resolveuid.py\n@@ -43,6 +43,7 @@ def uuidFor(obj):\n class ResolveUIDView(BrowserView):\n     """Resolve a URL like /resolveuid/<uuid> to a normalized URL."""\n \n+    uuid = None\n     subpath = None\n \n     def publishTraverse(self, request, name):\ndiff --git a/plone/outputfilters/tests/test_resolveuid_and_caption.py b/plone/outputfilters/tests/test_resolveuid_and_caption.py\nindex 91e8120..403d3ee 100644\n--- a/plone/outputfilters/tests/test_resolveuid_and_caption.py\n+++ b/plone/outputfilters/tests/test_resolveuid_and_caption.py\n@@ -299,6 +299,10 @@ def test_resolveuid_view_bad_uuid(self):\n         res = self.publish("/plone/resolveuid/BOGUS")\n         self.assertEqual(404, res.status)\n \n+    def test_resolveuid_view_missing_uuid(self):\n+        res = self.publish("/plone/resolveuid")\n+        self.assertEqual(404, res.status)\n+\n     def test_resolveuid_view_subpath(self):\n         res = self.publish("/plone/resolveuid/%s/image_thumb" % self.UID)\n         self.assertEqual(301, res.status)\n'

